pkg/plugin/system/system.go

```diff
@@ -217,14 +226,19 @@ type ClientProvider interface {
 	Kill()
 }
 
+type InternalPlugin struct {
```

Comment:
@JesseTatasciore I ended up doing something similar to what you did in your PR, to attach extra metadata to the plugin. I'm not sure which PR will merge first but let me know if there's anything here that doesn't align with yours or will cause annoying merge conflicts...

Reply:
I mean looking at what you changed, there will definitely be conflicts but I dont think it will be very hard to sort out

Reply:
I'm sure there will be conflicts but I'd hope mainly just with naming? Let me know if we're doing anything fundamentally different that would make the conflicts worse...